libs/threading_base/include/hpx/threading_base/external_timer.hpp

```diff
@@ -0,0 +1,332 @@
+//  Copyright (c) 2007-2016 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once    // prevent multiple inclusions of this header file.
+
+#include <hpx/config.hpp>
+#include <hpx/assertion.hpp>
+#include <hpx/coroutines/thread_id_type.hpp>
+#include <hpx/threading_base/thread_description.hpp>
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace hpx { namespace util {
+
+#ifdef HPX_HAVE_APEX
+
+    using enable_parent_task_handler_type = std::function<bool()>;
+
+    HPX_EXPORT void set_enable_parent_task_handler(
+        enable_parent_task_handler_type f);
+
+    namespace external_timer {
+
+        /* HPX provides a smart pointer to a data object that maintains
+     * information about an hpx_thread.  Any library (i.e. APEX)
+     * that wants to use this callback API needs to extend this class.
+     */
```

Comment:
This formatting looks wrong as well. Should we switch to using `//` style comments?

Reply:
Yikes. I think that's a good idea. I'll take care of the other formatting problems as well.